refactoring: dropdown component

maparr opened the pull request:
it closes #914 

from a refactoring perspective, cleaned up the dropdown component.

- [x] add to the Storybook the dropdown
- [x] add themes to the Storybook
- [x] replace all dropdowns 
- [x] clean up the code

maparr committed:
feat: add switchNetwork  dropdown story

feat: add switchNetwork  dropdown story

feat: add dropdowns to storybook

feat: refactor dropdown component

maparr committed:
feat: replace all dropdown items and clean up code

chore: [skip ci] clean up the dropdown storybook item

chore: [skip ci] clean up the dropdown storybook item

chore: [skip ci] clean up the dropdown storybook item

brusherru reviewed (changes requested):
1. Ugly margin in two-line dropdown:
    <img width="436" alt="Screenshot 2022-06-24 at 15 32 42" src="https://user-images.githubusercontent.com/1897530/175540861-e89ed246-88ce-4cf1-814d-d153c1578dcb.png">

2.  I'd like to collect all these dropdowns on the same story with ability to change background from dark to light as well. I think it will be more useful to check them out easily on the same page ;)

3. `rowContentCentered` props seems useless. What I'm missing?

4. `whiteIcon ` props confuses me. We have `isDarkMode` to switch to black bg and white icon. Let's remove it?

5. `isDarkMode: true` && theme light
     <img width="436" alt="image" src="https://user-images.githubusercontent.com/1897530/175542263-11d8e904-d150-47d7-8292-b7f5f2c3bea2.png">

maparr committed:
fix:
- remove bgColor, rowContentCentered from DropDown.tsx
- clean up Dropdown component in Storybook
- made theme definition dark or white background mode from themeProvider

maparr commented:
Hey @brusherru , done with additional clean up , for storybook , we have to select the background + theme 

<img width="420" alt="Screenshot 2022-06-26 at 14 46 44" src="https://user-images.githubusercontent.com/54288612/175819757-3ad0d174-7a23-4544-931e-2764d27f3ec8.png">
<img width="305" alt="Screenshot 2022-06-26 at 14 46 36" src="https://user-images.githubusercontent.com/54288612/175819759-8da5cb24-24da-416b-a281-259f88620f4f.png">

spaces for 2 lines in dropdown I made look same in previous version, let me know if I need force to previous view design

maparr committed:
feat: add all dropdons style variants in story

maparr committed:
feat: combine all dropdowns in story

brusherru commented on app/basicComponents/DropDown.tsx:
```diff
@@ -260,6 +269,48 @@ const ItemsWrapper = styled.div<{
+  font-weight: ${({ isBold }) => (isBold ? 800 : 400)};
+  font-size: 12px;
+  text-transform: uppercase;
+  margin-top: 10px;
```
```suggestion
  margin-top: 6px;
```
<img width="568" alt="image" src="https://user-images.githubusercontent.com/1897530/176413308-3ee0f6b4-58fc-404d-b385-5422be4f2d0d.png">

brusherru commented on app/basicComponents/DropDown.tsx:
```diff
@@ -28,7 +28,7 @@ const Wrapper = styled.div<{
       },
       isLightSkin,
```
A very confusing name.
As far as I understood this is some kind of alternative look (that is dark in "modern" and "dark" themes, but is not used in the "light" theme). So `isDarkMode` sounds better :)
Or we may call it `alternative` / `isDark` mb with suffix `Look` — then it will be pretty clear property name 🤔

brusherru reviewed (changes requested):
In the rest, I like it 👍

maparr committed:
fix:
- rename isLightSkin to isDarkMode
- change margin top for 2 line dropdown item

brusherru reviewed (approved):
👍
